### ISSUE TITLE
texinfo indentation bug?

### ISSUE THREAD

**Issue by mantepse** [sphinx-doc/sphinx#3080]
when producing texinfo documentation from

```
    .. RUBRIC:: Differences between ``CoordFunctionSymb`` and callable
      symbolic expressions

    Callable symbolic expressions are defined directly from symbolic
```

I obtain

```
@subsubheading Differences between @code{CoordFunctionSymb} and callable
symbolic expressions


Callable symbolic expressions are defined directly from symbolic
```

Is this a bug in sphinx or is there too little indentation after before `symbolic expressions`?

**Comment by mantepse** [comment]
Sorry, I should have been more precise: the expected output for texinfo is

```
@subsubheading Differences between @code{CoordFunctionSymb} and callable symbolic expressions


Callable symbolic expressions are defined directly from symbolic
```

(it must be on one line)

**Comment by mantepse** [comment]
I would like to ask yet another question in the same spirit.  The sphinx source is

```
REFERENCES:

.. [Lee11] \J.M. Lee : *Introduction to Topological Manifolds*,
   2nd ed., Springer (New York) (2011).
.. [Lee13] \J.M. Lee : *Introduction to Smooth Manifolds*,
   2nd ed., Springer (New York) (2013)
.. [KN63] \S. Kobayashi & K. Nomizu : *Foundations of Differential Geometry*,
   vol. 1, Interscience Publishers (New York) (1963).
.. [Huybrechts05] \D. Huybrechts : *Complex Geometry*,
   Springer (Berlin) (2005).
```

which in texinfo becomes

```
REFERENCES:
@anchor{sage/manifolds/manifold lee11}@anchor{1f}@w{(Lee11)} 
J.M. Lee : @emph{Introduction to Topological Manifolds},
2nd ed., Springer (New York) (2011).
@anchor{sage/manifolds/manifold lee13}@anchor{20}@w{(Lee13)} 
J.M. Lee : @emph{Introduction to Smooth Manifolds},
2nd ed., Springer (New York) (2013)
@anchor{sage/manifolds/manifold kn63}@anchor{21}@w{(KN63)} 
S. Kobayashi & K. Nomizu : @emph{Foundations of Differential Geometry},
vol. 1, Interscience Publishers (New York) (1963).
@anchor{sage/manifolds/manifold huybrechts05}@anchor{22}@w{(Huybrechts05)} 
D. Huybrechts : @emph{Complex Geometry},
Springer (Berlin) (2005).
```

which in turn is the following mess in the final info-manual:

```
REFERENCES: (Lee11) J.M. Lee : `Introduction to Topological Manifolds',
2nd ed., Springer (New York) (2011).  (Lee13) J.M. Lee : `Introduction
to Smooth Manifolds', 2nd ed., Springer (New York) (2013) (KN63) S.
Kobayashi & K. Nomizu : `Foundations of Differential Geometry', vol.  1,
Interscience Publishers (New York) (1963).  (Huybrechts05) D. Huybrechts
: `Complex Geometry', Springer (Berlin) (2005).
```

The same thing looks OK in html (each reference on one line, all decorated by a vertical bar on the left).  So, the question is as above: is the sphinx source bad, or does the texinfo-builder need improvements.

If the answer is the latter, could you give me a hint, where to look (I suppose I should tweak `writers/texinfo.py`, but which method?

**Comment by jfbu** [comment]
for info: it looks also ok in latex PDF output

**Comment by mantepse** [comment]
I found out how to solve the first problem.  In `writers/texinfo.py`, one has to use `escape_newlines`:

```
 def visit_rubric(self, node):
        self.escape_newlines += 1
        if len(node.children) == 1 and node.children[0].astext() in \
                ('Footnotes', _('Footnotes')):
            raise nodes.SkipNode
        try:
            rubric = self.rubrics[self.section_level]
        except IndexError:
            rubric = self.rubrics[-1]
        self.body.append('\n%s ' % rubric)

    def depart_rubric(self, node):
        self.body.append('\n\n')
        self.escape_newlines -= 1   
```

I think one should do the same also for `visit_title`, but I didn't encounter such a case yet.

**Comment by mantepse** [comment]
Hm, not quite, because now there are to few newlines elsewhere :-(
(in the `example` blocks)

**Comment by mantepse** [comment]
there is a patch at https://trac.sagemath.org/ticket/21734 now.  However, it still needs work.

**Comment by mantepse** [comment]
I updated the branch on https://trac.sagemath.org/ticket/21734 so that it's easily visible, but its been too long, so I cannot tell anymore what remains to be done...

**Comment by mantepse** [comment]
I just checked, the blocker to enable a texinfo build for sagemath is issue #3089 (which has been closed because too difficult to fix).

**Comment by tk0miya** [comment]
Fixed by #5600.
Thank you for reporting.
